Microsoft.HealthVault.Fhir/Transformers/BloodPressureToFhir.cs

```diff
@@ -0,0 +1,65 @@
+//
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+using System.Collections.Generic;
+using System.Linq;
+using Hl7.Fhir.Model;
+using Microsoft.HealthVault.Fhir.Codes.HealthVault;
+using Microsoft.HealthVault.Fhir.Constants;
+using Microsoft.HealthVault.ItemTypes;
+
+namespace Microsoft.HealthVault.Fhir.Transformers
+{
+    public static partial class ThingBaseToFhir
+    {
+        // Register the type on the generic ThingToFhir partial class
+        public static Observation ToFhir(this BloodPressure bp)
+        {
+            return BloodPressureToFhir.ToFhirInternal(bp, ThingBaseToFhir.ToFhirInternal(bp));
+        }
+    }
+
+    internal static class BloodPressureToFhir
+    {
+        internal static Observation ToFhirInternal(BloodPressure bp, Observation observation)
+        {
+            if (bp.IrregularHeartbeatDetected.HasValue)
+            {
+                observation.AddExtension(HealthVaultVocabularies.IrregularHeartBeatExtensionName, new FhirBoolean(bp.IrregularHeartbeatDetected.Value));                
+            }
+            
+            var diastolicComponent = new Observation.ComponentComponent
+            {
+                Code = new CodeableConcept() { Coding = new List<Coding> { HealthVaultVitalStatisticsCodes.BloodPressureDiastolic } },
+                Value = new Quantity((decimal)bp.Diastolic, "mmHg")
+            };
+
+            var systolicComponent = new Observation.ComponentComponent
+            {
+                Code = new CodeableConcept() { Coding = new List<Coding> { HealthVaultVitalStatisticsCodes.BloodPressureSystolic } },
+                Value = new Quantity((decimal)bp.Systolic, "mmHg")
+            };
+
+            observation.Component = new List<Observation.ComponentComponent> { diastolicComponent, systolicComponent };
+
+            if (bp.Pulse != null)
+            {
+                observation.Component.Add(new Observation.ComponentComponent
+                {
+                    // Code = new CodeableConcept() { Coding = new List<Coding> { HealthVaultVitalStatisticsCodes.HeartRate } },
```

Comment:
Is there any reason for this to be commented out instead of either implemented or removed?